spring-xd-spark-streaming/src/main/java/org/springframework/xd/spark/streaming/java/ModuleExecutor.java

```diff
@@ -72,7 +73,10 @@ public void call(Iterator<?> results) throws Exception {
 									}
 								}
 								while (results.hasNext()) {
-									messageSender.send(MessageBuilder.withPayload(results.next()).build());
+									Object next = results.next();
+									Message message = (next instanceof Message) ? (Message) next :
+											MessageBuilder.withPayload(next).build();
```

Comment:
Not critical for this release but we should always use `getMessageBuilderFactory(beanFactory)` to get the `MessageBuilder` to use (not each time of course, during initialization).

This allows customization of the message builder globally.